doc/decisions/warning-array.md

```diff
@@ -0,0 +1,36 @@
+# Arrays
+
+## Problem
+
+Currently multiple warnings are saved in an elektra non-conforming array
+notation which is limited to 100 entries. The notation of `#00` is against
+the design [decision made](array.md).
+
+Furthermore if the number of warnings is exceeded, the earliest entries are overwritten
+which might yield the most useful warning messages.
+
+## Constraints
+
+- The number of entries should stay limited but may be more than 100 entries.
+
+## Assumptions
+
+## Considered Alternatives
+
+## Decision
+
+The format should be aligned with the correct array notation,
+starting with `#0`. The maximum number of warnings will be expanded to
+1001 entries (`#0` - `#1000`).
```

Comment:
Actually, there is no reason to change the current value of 100, nobody complained that it is too low. It only should not be hard-coded  that it must be 100.

```suggestion
1001 entries (`#0` - `#___1000`).
```

Reply:
We agreed in the elektra meeting to hardcap the limit in case of recursive functions which issue warnings and never end.

Reply:
Yes, we agreed on that. But not on a concrete number.

If you give any reason for why it should be 1000, it might be ok.

Reply:
Actually I remember that we agreed on the number 1000 but any other number is fine too. But I think we need a hard-cap somewhere to avoid filling up the whole system with warning messages if someone accidentally introduces an endless loop.